core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -89,17 +89,29 @@ protected void killComputer(Computer c) {
         c.kill();
     }
 
-    /* =================================================================================================================
-    * Package-protected, but accessed API
-    * ============================================================================================================== */
+    private final Set<String> disabledAdministrativeMonitors = Collections.synchronizedSet(new HashSet<>());
 
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    /**
+     * Get the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public Set<String> getDisabledAdministrativeMonitors(){
+        return disabledAdministrativeMonitors;
+    }
 
-    @Restricted(NoExternalUse.class)
-    public CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
-    	return disabledAdministrativeMonitors;
+    /**
+     * Set the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public synchronized void setDisabledAdministrativeMonitors(Set<String> disabledAdministrativeMonitors) {
+        synchronized (this.disabledAdministrativeMonitors) {
```

Comment:
Is it really needed to `synchronized` method and inside method?

Reply:
Looks like I missed that. The method doesn't need to be synchronized any more (and furthermore, the `synchronized` on the method is using a different lock than the one we're interested in).